webapp/portlet/src/main/webapp/vue-apps/search/components/SearchResults.vue

```diff
@@ -290,16 +290,39 @@ export default {
         return;
       }
       if (!selectedConnector.enabled || this.connectors.length === this.enabledConnectors.length) {
-        const searchFilterAnalytics = {
+        let name;
+        switch (selectedConnector.name) {
+        case 'activity':
+          name = 'searchActivities';
+          break;
+        case 'news':
+          name = 'searchNews';
+          break;
+        case 'wiki':
+          name = 'searchNotes';
+          break;
+        case 'people':
+          name = 'searchPeople';
+          break;
+        case 'perkstore':
+          name = 'searchProducts';
+          break;
+        case 'agenda':
+          name = 'searchEvents';
+          break;
+        default:
+          name = `search${selectedConnector.name.charAt(0).toUpperCase()}${selectedConnector.name.slice(1)}s`;
+        }
+        const searchFilter = {
           'module': 'portal',
           'subModule': 'ui',
           'userId': eXo.env.portal.userIdentityId,
           'userName': eXo.env.portal.userName,
           'operation': 'click',
-          'name': `search${selectedConnector.name.charAt(0).toUpperCase()}${selectedConnector.name.slice(1)}`,
+          'name': name,
           'timestamp': Date.now()
         };
-        document.dispatchEvent(new CustomEvent('exo-statistic-message', {detail: searchFilterAnalytics}));
+        document.dispatchEvent(new CustomEvent('exo-statistic-message', {detail: searchFilter}));
```

Comment:
We can't add statistics collection here, all you can do is to trigger a generic event that should be catched by eXo Analytics addon to collect statistics (No dependency from social to Analytics should be visible in source code of Meeds).

```suggestion
        document.dispatchEvent(new CustomEvent('search-filter-selected', {detail: selectedConnector.name}));
```

And we should delete all the upper computation that builds an analytics object `searchFilter` which should be made in Analytics addon only.

Reply:
Updated !